src/scripts/main.js

```diff
@@ -1,3 +1,67 @@
 'use strict';
 
 // write code here
+
+const rowAppended = document.querySelector('.append-row');
+const rowRemover = document.querySelector('.remove-row');
+const columnAppended = document.querySelector('.append-column');
+const columnRemover = document.querySelector('.remove-column');
+const field = document.querySelector('.field');
+
+function updateButtonStates() {
+  const trs = field.querySelectorAll('tr');
+  const tds = trs.length > 0 ? trs[0].children : [];
+
+  rowAppended.disabled = trs.length >= 10;
+  rowRemover.disabled = trs.length <= 2;
+  columnAppended.disabled = tds.length >= 10;
+  columnRemover.disabled = tds.length <= 2;
+}
+
+updateButtonStates();
+
+rowAppended.addEventListener('click', (e) => {
+  const trs = field.querySelectorAll('tr');
```

Comment:
Consider using `field.rows` instead of `querySelectorAll('tr')` for better performance and readability, as `field.rows` directly provides a live HTMLCollection of the rows.